core/store/src/lib.rs

```diff
@@ -296,7 +296,7 @@ pub fn read_with_cache<'a, T: BorshDeserialize + 'a>(
     Ok(None)
 }
 
-pub fn create_store(path: &str) -> Arc<Store> {
+pub fn create_store<P: AsRef<std::path::Path>>(path: P) -> Arc<Store> {
```

Comment:
```suggestion
pub fn create_store(path: &Path) -> Arc<Store> {
```

See [this rationale](https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#avoid-monomorphization).

For super-public API, `AsRef<Path>` might make sense, but should be written by delegating to `&Path` function, the way std does it:

```rust
pub fn read_to_string<P: AsRef<Path>>(path: P) -> io::Result<String> {
    fn inner(path: &Path) -> io::Result<String> {
        let mut file = File::open(path)?;
        let mut string = String::with_capacity(initial_buffer_size(&file));
        file.read_to_string(&mut string)?;
        Ok(string)
    }
    inner(path.as_ref())
}
```

Reply:
thanks for detailed explanation, fixed